src/Graphing.hs

```diff
@@ -92,13 +92,17 @@ edge parent child gr = gr { graphingAdjacent = adjacent' }
 -- - A way to convert a dependency @toDependency@
 --
 -- __Unfold does not work for recursive inputs__
-unfold :: Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
-unfold seed getDeps toDependency = foldr addNode empty seed
+unfold :: forall dep res. Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
+unfold seed getDeps toDependency = Graphing
+  { graphingDirect = S.fromList (map toDependency seed)
+  , graphingAdjacent = AM.vertices (map toDependency seed) `AM.overlay` AM.edges [(toDependency parentDep, toDependency childDep) | (parentDep,childDep) <- edgesFrom seed]
+  }
   where
-  addNode dep gr = direct res (foldr (edge res . toDependency) gr children)
-    where
-    children = getDeps dep
-    res = toDependency dep
+    edgesFrom :: [dep] -> [(dep,dep)]
+    edgesFrom nodes = do
+      node <- nodes
```

Comment:
[nit]: This is probably a little confusing to most readers, including me.  I had to really think about how this would work.  I don't know how many people will read this AND need to fully understand it AND not be able to figure out like I did, but it may be worth writing the iteration more explicitly.

Reply:
I agree it's likely to be confusing the first few times it's encountered -- but I think the alternatives are potentially more-confusing.

Having an intuition for how the list monad works (or: "not every monad is a container with one thing in it") is useful in other places too, especially for effects like `NonDet` in fused-effects, or reasoning about megaparsec/etc `Parser` monads. I wish there was a better way to signal that "this is a list do block"

Reply:
I'm on the fence here.  I marked this as a nit, because it shouldn't block the PR, but I'm not convinced that this is the most readable syntax for what is essentially a nested map pipeline.  But this code may not change, so no need to block it here.